Lib/py_compile.py

```diff
@@ -137,6 +137,10 @@ def compile(file, cfile=None, dfile=None, doraise=False, optimize=-1):
     except FileExistsError:
         pass
     source_stats = loader.path_stats(file)
+    sde = os.environ.get('SOURCE_DATE_EPOCH')
```

Comment:
In Debian, we don't ship binary packages with pyc files, instead we generate them when the package is installed.  That makes more sense for us because we can support multiple versions of Python simultaneously, so we generate pycs based on the versions installed on the target system.  That way we generally don't have to build all of our binary packages when we add a new Python interpreter, and we don't bloat our binary packages with pyc files.

But in any case, `$SOURCE_DATE_EPOCH` is the de facto standard for transmitting this to programs for reproducible builds, so this change is reasonable.  However I would like to see documentation and a test.